### ISSUE TITLE
test_config_parser.cpp:(.text+<snip>): undefined reference to Davix::davix_config_tokenize(...)

### ISSUE THREAD

**Issue by amadio** [cern-it-sdc-id/davix#24]
Davix has been recently packaged in Gentoo, but it is failing to build with testing turned on, as shown below. For more information, please see https://bugs.gentoo.org/651098.

```
CMakeFiles/g_test_unit_davix_exe.dir/parser/test_config_parser.cpp.o: In function `ConfigParser_Tokenizer_Test::TestBody()':
test_config_parser.cpp:(.text+0x22f): undefined reference to `Davix::davix_config_tokenize(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&)'
test_config_parser.cpp:(.text+0x576): undefined reference to `Davix::davix_config_tokenize(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&)'
test_config_parser.cpp:(.text+0x804): undefined reference to `Davix::davix_config_tokenize(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&)'
test_config_parser.cpp:(.text+0xc02): undefined reference to `Davix::davix_config_tokenize(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&)'

  -------------------------------------------------------------------

  This is an unstable amd64 chroot image at a tinderbox (==build bot)
  name: 17.0-developer_20180319-191208

  -------------------------------------------------------------------

gcc-config -l:
 [1] x86_64-pc-linux-gnu-7.3.0 *

Available Python interpreters, in order of preference:
  [1]   python3.5
  [2]   python2.7 (fallback)
```

**Comment by gbitzes** [comment]
Hi, thanks for the report! I pushed a fix on devel branch, could you give it a try?

**Comment by amadio** [comment]
Hi @gbitzes. Thanks for the quick response. Looking at your patch I realized that the problem is that to have `-DUNIT_TEST=ON`, you also need `-DENABLE_TOOLS=ON`, otherwise (even with your patch), things do not work. I added in Gentoo the requirement that when testing is enabled the tools also have to be enabled. In that case, all tests pass even without your change. In any case, thank you for your help! Cheers,
